private/buf/cmd/buf/command/push/push.go

```diff
@@ -199,6 +220,33 @@ func push(
 	if err != nil {
 		return nil, err
 	}
+	modulePin, err := push(ctx, container, clientConfig, moduleIdentity, builtModule, flags)
+	if err != nil {
+		// We rely on Push* returning a NotFound error to denote the repository is not created.
+		// This technically could be a NotFound error for some other entity than the repository
+		// in question, however if it is, then this Create call will just fail as the repository
+		// is already created, and there is no side effect. The 99% case is that a NotFound
+		// error is because the repository does not exist, and we want to avoide having to do
+		// a GetRepository RPC call for every call to push --create.
+		if flags.Create && connect.CodeOf(err) == connect.CodeNotFound {
+			if err := create(ctx, container, clientConfig, moduleIdentity, flags); err != nil {
```

Comment:
I just realized one other annoying thing: in the case the comment refers to, or on a random race condition, where this create call is done in error on a repository that already exists, this will return a CodeAlreadyExists, which in turn will result in a message to stderr that the commit already exists and a zero exit code. This is wrong - in this case, we should error, with a message that stating we expected the repository to not exist but it did, and this should be changed to an internal error.

Reply:
Yeah, that's reasonable -- I can make that check when the error is checked.